appserver/web/war-util/src/main/java/com/sun/enterprise/glassfish/web/WarHandler.java

```diff
@@ -263,10 +262,10 @@ protected void configureLoaderProperties(WebappClassLoader cloader, WebXmlParser
                     // support exploded jar file
                     cloader.addRepository("WEB-INF/lib/" + file.getName() + "/", file);
                 } else {
-                    cloader.addJar(file.getPath().substring(baseFileLen), new JarFile(file), file);
+                    cloader.addJar(file.getPath().substring(baseFileLen), file);
                 }
             } catch (final Exception e) {
-                // Catch and ignore any exception in case the JAR file is empty.
+                LOG.log(Level.FINEST, "Could not add file " + file, e);
```

Comment:
No need to change, but just a remark: many places use a static import for FINEST.

Reply:
True, but not in WarHandler, I checked it now for sure. I tried to limit changes outside of loader, despite I think there is one or two classes where I "failed" and committed some eclipse autosave changes (foreach, override, braces).